File: core/src/main/java/dev/wesley/shooter/ShooterGame.java
package dev.wesley.shooter;

import com.badlogic.gdx.Screen;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import dev.wesley.shooter.screens.UIScreen;
import ktx.app.KtxGame;

/** {@link com.badlogic.gdx.ApplicationListener} implementation shared by all platforms. */
public class ShooterGame extends KtxGame<Screen> {
	private SpriteBatch batch;
	private Texture image;

	@Override
	public void create() {
		batch = new SpriteBatch();
		image = new Texture("libgdx.png");
		addScreen(UIScreen.class, new UIScreen());
		setScreen(UIScreen.class);
		super.create();
	}

	@Override
	public void render() {
		super.render();
//		Gdx.gl.glClearColor(0.15f, 0.15f, 0.2f, 1f);
//		Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
//		batch.begin();
//		batch.draw(image, 140, 210);
//		batch.end();
	}

	@Override
	public void dispose() {
		batch.dispose();
		image.dispose();
		super.dispose();
	}
}

File: core/src/main/java/dev/wesley/shooter/ShooterGame.kt
package dev.wesley.shooter

import com.badlogic.gdx.Application
import com.badlogic.gdx.ApplicationListener
import com.badlogic.gdx.Gdx
import dev.wesley.shooter.screen.GameScreen
import ktx.app.KtxGame
import ktx.app.KtxScreen

/** [com.badlogic.gdx.ApplicationListener] implementation shared by all platforms. */
class ShooterGame : KtxGame<KtxScreen>(), ApplicationListener {

    /**
     * Create method which renders the screen.
     */
    override fun create() {
        Gdx.app.logLevel = Application.LOG_DEBUG
        addScreen(GameScreen())
        setScreen<GameScreen>()
    }

    companion object {
        const val UNIT_SCALE = 1/16f

    }

}
